internal/gatewayapi/servicecluster.go

```diff
@@ -0,0 +1,88 @@
+// Copyright Envoy Gateway Authors
+// SPDX-License-Identifier: Apache-2.0
+// The full text of the Apache license is available in the LICENSE file at
+// the root of the repo.
+
+package gatewayapi
+
+import (
+	"fmt"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/utils/ptr"
+
+	"github.com/envoyproxy/gateway/internal/envoygateway/config"
+	"github.com/envoyproxy/gateway/internal/gatewayapi/resource"
+	"github.com/envoyproxy/gateway/internal/ir"
+	"github.com/envoyproxy/gateway/internal/utils"
+)
+
+func (t *Translator) ProcessServiceCluster(acceptedGateways []*GatewayContext, resources *resource.Resources, xdsIR resource.XdsIRMap) {
```

Comment:
can we move this under https://github.com/envoyproxy/gateway/blob/ceec66655765bea49635f951fa075796ca47796d/internal/gatewayapi/globalresources.go#L21

it loops across the IRs which accounts for the service cluster name in the IR key itself so we dont need any additional merge gateway or service cluster name logic

```
svc := resources.GetService(t.ControllerNamespace, svcName)
		if svc == nil {
			return
		}
```

Reply:
The comment was around avoiding redefining this logic, and is superceded by this comment 
https://github.com/envoyproxy/gateway/pull/6597#discussion_r2237026272

Ideally we should be able to populate this new  IR subfield without redfining core logic